pyfar/dsp/dsp.py

```diff
@@ -1929,6 +1930,19 @@ def average(signal, mode='linear', caxis=None, weights=None, keepdims=False):
         If this is ``True``, the axes which are reduced during the averaging
         are kept as a dimension with size one. Otherwise, singular dimensions
         will be squeezed after averaging. The default is ``False``.
+    nan_policy: string, optional
+        Define how to handle NaNs in input signal.
+
+        ``'propagate'``
+           If the input signal includes NaNs, the resulting averaged output
+           signal values are NaN.
```

Comment:
This could maybe be formulated a bit more precise. Not all values in the resulting signal are NaN, only at the corresponding samples or bins, right?